modules/library/main/src/main/java/org/geotools/filter/function/math/FilterFunction_IEEEremainder.java

```diff
@@ -40,25 +42,29 @@ public FilterFunction_IEEEremainder() {
     }
 
     public Object evaluate(Object feature) {
-        double arg0;
-        double arg1;
 
-        try { // attempt to get value and perform conversion
-            arg0 = getExpression(0).evaluate(feature, Double.class).doubleValue();
-        } catch (Exception e) {
+        Object arg0 = getExpression(0).evaluate(feature);
+        Object arg1 = getExpression(1).evaluate(feature);
+        if (arg0 == null || arg1 == null) {
+            return null;
+        }
+        Hints hints0 = new Hints();
+        arg0 =
```

Comment:
No need to pass down hints as a third argument, use the two argument method instead